09-maze_solver/progress/internal/solver/solver.go

```diff
@@ -0,0 +1,77 @@
+package solver
+
+import (
+	"fmt"
+	"image"
+	"image/gif"
+	"log/slog"
+)
+
+// Solver is capable of finding the path from the entrance to the treasure.
+// The maze has to be a RGBA image.
+type Solver struct {
+	maze   *image.RGBA
```

Comment:
It would be better to use correctly colored `image.Paletted` instead. i.e. you assign a palette number to each color.

This would also avoid checking against a RGBA color and instead a palette index.

Reply:
We decided to keep `image.RGBA` but we mentioned `image.Paletted` in the chapter.

Reply:
Note, the main reason I was suggesting paletted, was because it'll make the code nicer and faster overall.

Reply:
Here's quick example how it's possible to use image.Paletted directly https://gist.github.com/egonelbre/1e453f716d63ccba134f4c5e5dcd7a5c